### PR TITLE
Create basic db model

### PR DESCRIPTION
**Pull request by returnv01d**
Created and connected all models except 'FoodRequest' with fresh 'DietIngredient' instead of 'DietMeals' connection (TY @Qsmir :heart:  :rocket: :rocket: :rocket: )

### CODE REVIEW

File: app/models/diet.rb [hunk 1]
```diff
@@ -1,2 +1,10 @@
 class Diet < ApplicationRecord
+  has_and_belongs_to_many :users,
+                          class_name: 'User' ,
```
Comment (Lemcur) [inline app/models/diet.rb]: Idk if it's needed to call class_name 'User' when it's :users who habtm

File: app/models/ingredient.rb [hunk 2]
```diff
@@ -2,5 +2,13 @@ class Ingredient < ApplicationRecord
   has_and_belongs_to_many :allergic_users,
                           class_name: 'User' ,
```
Comment (Lemcur) [inline app/models/ingredient.rb]: here it's mandatory tho

File: db/migrate/20180827103128_create_meals_ingredients.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateMealsIngredients < ActiveRecord::Migration[5.2]
+  def change
+    create_table :meals_ingredients do |t|
+
```
Comment (Lemcur) [inline db/migrate/20180827103128_create_meals_ingredients.rb]: Shouldn't it be a join table? We've got only one index here. I ask a lot of questions. That's not how reviews should be done x.x

File: db/schema.rb [hunk 4]
```diff
@@ -1,19 +1,82 @@
-ActiveRecord::Schema.define(version: 2018_08_24_120902) do
+# This file is auto-generated from the current state of the database. Instead
```
Comment (Lesin) [inline db/schema.rb]: Let's remove those comments ;)

File: db/schema.rb [hunk 5]
```diff
@@ -1,19 +1,82 @@
...
+#
+# It's strongly recommended that you check this file into your version control system.
 
+ActiveRecord::Schema.define(version: 2018_08_27_135647) do
+
+  # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
   create_table "caterings", force: :cascade do |t|
-    t.string "name", null: false
```
Comment (Lesin) [inline db/schema.rb]: Could catering name be empty?

File: db/schema.rb [hunk 6]
```diff
@@ -1,19 +1,82 @@
...
+  end
+
+  create_table "ingredients", force: :cascade do |t|
+    t.string "name"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
   create_table "meals", force: :cascade do |t|
-    t.string "name", null: false
```
Comment (Lesin) [inline db/schema.rb]: Can meal name be empty? :)

Comment (Lemcur) [reply db/schema.rb]: thank You <3

File: app/models/diet.rb [hunk 7]
```diff
@@ -1,2 +1,10 @@
...
+                          foreign_key: 'diet_id',
+                          join_table: :users_diets
+  has_and_belongs_to_many :ingredients,
+                          class_name: 'Ingredient' ,
+                          foreign_key: 'diet_id',
+                          join_table: :diets_indegrients
```
Comment (pgorni) [inline app/models/diet.rb]: ❌ indegrients
⭕ ingredients

File: db/migrate/20180827103128_create_meals_ingredients.rb (hunk 3 above)
Comment (pgorni) [reply db/migrate/20180827103128_create_meals_ingredients.rb]: I believe it should, just like UsersAllergies.
It should probably have two links: `meal_id` and `ingredient_id`.

Comment (returnv01d) [reply db/migrate/20180827103128_create_meals_ingredients.rb]: yes, it should have 'meal' connection too

File: app/models/diet.rb (hunk 7 above)
Comment (returnv01d) [reply app/models/diet.rb]: like with the adress thing :1st_place_medal:

File: app/models/diet.rb (hunk 1 above)
Comment (kjuri) [reply app/models/diet.rb]: true that :)

**Pull request merged**